pkg/config/authorization.go

```diff
@@ -20,14 +21,16 @@ type AuthorizationConfig struct {
 	KubernetesAuthz *authorization.KubernetesAuthz     `yaml:"kubernetes,omitempty"`
 }
 
-func (config *AuthorizationConfig) Call(pipeline common.AuthPipeline, ctx context.Context) (interface{}, error) {
+func (config *AuthorizationConfig) Call(pipeline common.AuthPipeline, ctx context.Context, parentLogger log.Logger) (interface{}, error) {
+	logger := parentLogger.WithName("authorization")
+
 	switch {
 	case config.OPA != nil:
-		return config.OPA.Call(pipeline, ctx)
+		return config.OPA.Call(pipeline, ctx, logger)
```

Comment:
@guicassolato  another option would be to inject the logger in the context. 

Check https://github.com/kubernetes-sigs/controller-runtime/blob/1462c6e3a5dbe3b04d11286b190253c845c8e338/pkg/log/log.go#L85
```
func FromContext(ctx context.Context, keysAndValues ...interface{}) logr.Logger
```

and 

```
func IntoContext(ctx context.Context, log logr.Logger) context.Context
```